Django-CoWork-Connect/coapp/admin.py

```diff
@@ -1,8 +1,30 @@
 from django.contrib import admin
-from .models import Desk
+from .models import Desk, Reservation
+
 
-@admin.register(Desk)
 class DeskAdmin(admin.ModelAdmin):
     list_display = ('stock_number', 'size', 'monitor_number', 'power_socket_count', 'price', 'status')
     list_filter = ('monitor_number', 'size', 'status')
-    search_fields = ['stock_number']
\ No newline at end of file
+    search_fields = ['stock_number', 'status']
+    actions = ['cancel_reservations']
+
+
+class ReservationAdmin(admin.ModelAdmin):
+    list_display = ('desk', 'id', 'user', 'start_date', 'end_date', 'total_cost')
+    list_filter = ('start_date', 'end_date', 'user')
+    search_fields = ('id', 'user__username', 'desk__name')
+    raw_id_fields = ('desk', 'user')
+
+    def cancel_reservations(self, request, queryset):
+        for reservation in queryset:
+            desk = reservation.desk
+            desk.status = 'czynne'
+            desk.save()
+            reservation.delete()
+        self.message_user(request, "Wybrane rezerwacje zostały anulowane i statusy biurek zaktualizowane.")
```

Comment:
albo jescze prosciej ale nie jestem pewny więc sppawdz :) 
```suggestion
        queryset.update(desk__status='czynne')
        queryset.delete()
        
        EWENTUALNIE TAK ALE OBU NIE JESTEM PEWNY TO NA W KOMENTARZU WYZEJ NA BANK ZADZIALA
        from django.db.models import F, Value
        queryset.update(desk__status=Value('czynne'))
        queryset.delete()
```

Reply:
ok.